bin/nrnivmodl_makefile_cmake.in

```diff
@@ -68,9 +68,9 @@ C_GREEN := \033[32m
 # Take the main and link with nrnmech.
 # RPATH is set for DESTDIR_RPATH and coreneuron lib
 special: $(mech_lib)
-	@printf " => $(C_GREEN)LINKING$(C_RESET) executable $(special) LDFLAGSE are: $(LDFLAGS)\n"
-	(cd .. ;  $(CXX_LINK_EXE) -I $(incdir) -I $(incdir)/nrncvode $(datadir)/nrnmain.cpp -o $(special) \
-	  $(mod_func_o) -L $(OBJS_DIR) -l$(mech_libname) $(NRNLIB_FLAGS) -Wl,-rpath,'$(DESTDIR_RPATH)' -Wl,-rpath,$(libdir) $(LDFLAGS))
+	@printf " => $(C_GREEN)LINKING$(C_RESET) executable $(special) LDFLAGS are: $(LDFLAGS)\n"
+	(cd .. ;  $(CXX_LINK_EXE) -I $(incdir) -I $(incdir)/nrncvode -DAUTO_DLOPEN_NRNMECH=0 $(datadir)/nrnmain.cpp -o $(special) \
+	  -L $(OBJS_DIR) -l$(mech_libname) $(NRNLIB_FLAGS) -l$(mech_libname) -Wl,-rpath,'$(DESTDIR_RPATH)' -Wl,-rpath,$(libdir) $(LDFLAGS))
```

Comment:
duplicated the -lnrnmech because of mutual references in libnrniv.so otherwise prevent linking. I suppose that was the original reason for linking modfunc.o and maybe that was better but modfunc.o is already in libnrnmech.so

Reply:
Was wondering as well why modfunc.o and mod_objs were needed. Funny enough, it was working without them on my mac , but then re-added them one by one because travis was failing.

Reply:
This is related to the old meaning of NRN_ENABLE_MECH_DLL_STYLE, or more precisely the
linux_nrnmech autotools variable. In that case there were no libnrnmech.so dynamic libraries and so the special was  a main + all the mod files linked statically.  This predated the ability to
launch python. Now that is useless if one launches python. Of course one can still embed python in the libnrniv.so. Also libnrniv can be built statically as a libnrniv.a and linked statically into the special.  One gets a choice at that point whether to embed python dynamically or, if a libpython.a exists, to link that statically as well. Anyway, none of that is determined by
-DNRN_ENABLE_BINARY_SPECIAL=NO